keys.go

```diff
@@ -260,6 +269,41 @@ func (c *Client) SetKeyRing(ctx context.Context, idOrAlias, newKeyRingID string)
 	return &response.Keys[0], nil
 }
 
+// CreateKeyWithPolicyOverride creates a new KP key with given policy details
+func (c *Client) CreateKeyWithPolicyOverride(ctx context.Context, name string, expiration *time.Time, extractable bool, aliases []string, policy Policy) (*Key, error) {
+	return c.CreateImportedKeyWithPolicyOverride(ctx, name, expiration, "", "", "", extractable, aliases, policy)
+}
+
+// CreateImportedKeyWithPolicyOverride creates a new Imported KP key from the given key material and with given DualAuthDelete policy details
+func (c *Client) CreateImportedKeyWithPolicyOverride(ctx context.Context, name string, expiration *time.Time, payload, encryptedNonce, iv string, extractable bool, aliases []string, policy Policy) (*Key, error) {
+	key := c.createKeyTemplate(ctx, name, expiration, payload, encryptedNonce, iv, extractable, aliases, AlgorithmRSAOAEP256)
+
+	key.Rotation = policy.Rotation
+	key.DualAuthDelete = policy.DualAuth
+
+	return c.createKeyWithPolicyOverride(ctx, key)
+}
+
+// CreateRootKeyWithPolicyOverride creates a new, non-extractable key resource without key material and with given key policy details
+func (c *Client) CreateRootKeyWithPolicyOverride(ctx context.Context, name string, expiration *time.Time, aliases []string, policy Policy) (*Key, error) {
+	return c.CreateKeyWithPolicyOverride(ctx, name, expiration, false, aliases, policy)
+}
+
+// CreateStandardKeyWithPolicyOverride creates a new, extractable key resource without key material and with given DualAuthDelete policy details
+func (c *Client) CreateStandardKeyWithPolicyOverride(ctx context.Context, name string, expiration *time.Time, aliases []string, policy Policy) (*Key, error) {
+	return c.CreateKeyWithPolicyOverride(ctx, name, expiration, true, aliases, policy)
+}
+
+// CreateImportedRootKey creates a new, non-extractable key resource with the given key material and with given DualAuthDelete policy details
+func (c *Client) CreateImportedRootKeyWithPolicyOverride(ctx context.Context, name string, expiration *time.Time, payload, encryptedNonce, iv string, aliases []string, policy Policy) (*Key, error) {
+	return c.CreateImportedKeyWithPolicyOverride(ctx, name, expiration, payload, encryptedNonce, iv, false, aliases, policy)
+}
+
+// CreateStandardKey creates a new, extractable key resource with the given key material and with given DualAuthDelete policy details
```

Comment:
```suggestion
// CreateImportedStandardKeyWithPolicyOverride creates a new, extractable key resource with the given key material and with given key policy details
```

Reply:
for above case, we know that if we pass rotation info in the key body and we return the error.

But I think the function description has be general only which includes both the rotation and DA key policies.

@dinesh-venkatraman-g any thoughts?

Reply:
Yeah, In my opinion the sdk's job is to just carry the payload correctly to the REST API for any syntactical or logical error, the API should handle those cases.